packages/langium/src/workspace/configuration.ts

```diff
@@ -77,34 +87,25 @@ export class DefaultConfigurationProvider implements ConfigurationProvider {
                 });
             }
 
-            // params.fetchConfiguration(...) is a function to be provided by the calling language server for the sake of
-            //  decoupling this implementation from the concrete LSP implementations, specifically the LSP Connection
-            this.fetchConfiguration = params.fetchConfiguration;
-
-            // awaiting the fetch of the initial configuration data must not happen here, because it would block the
-            //  initialization of the language server, and may lead to out-of-order processing of subsequent messages from the language client;
-            // fetching the initial configuration in a non-blocking manner might cause a read-before-write problem
-            //  in case the workspace initialization relies on that configuration data (which is the case for the grammar language server);
-            // consequently, we fetch the initial configuration data on blocking manner on-demand, when the first configuration value is read,
-            //  see below in #getConfiguration(...);
+            await this.initializeConfiguration(params);
         }
+        this._ready.resolve();
     }
 
-    protected async initializeConfiguration(): Promise<void> {
-        if (this.fetchConfiguration) {
+    protected async initializeConfiguration(params: ConfigurationInitializedParams): Promise<void> {
```

Comment:
If you implement it this way you then could merge the two methods, i.e. inline the content of `initializeConfiguration()`, or do you expect any need for the latter as a customization hook?